com.unity.multiplayer.mlapi/Runtime/Spawning/NetworkSpawnManager.cs

```diff
@@ -601,26 +601,19 @@ internal void ClientCollectSoftSyncSceneObjectSweep(NetworkObject[] networkObjec
             }
         }
 
-        internal void OnDestroyObject(ulong networkId, bool destroyGameObject)
+        internal void OnDestroyObject(NetworkObject sobj, bool destroyGameObject)
         {
             if (NetworkManager == null)
             {
                 return;
             }
 
-            //Removal of spawned object
-            if (!SpawnedObjects.TryGetValue(networkId, out NetworkObject sobj))
-            {
-                Debug.LogWarning($"Trying to destroy object {networkId} but it doesn't seem to exist anymore!");
-                return;
-            }
-
```

Comment:
We may be able to remove this block but in its place we should do a null check on the passed in network object

Reply:
I think we should assert for the following here:
- NotNull
- NetworkObject.IsSpawned
- SpawnedObjects.Contains(sobj)

That way we definitely catch if anything is wrong in a debug build without affecting release build performance.